pkg/koordlet/statesinformer/impl/states_pods.go

```diff
@@ -141,6 +145,70 @@ func (s *podsInformer) GetAllPods() []*statesinformer.PodMeta {
 	return pods
 }
 
+func (s *podsInformer) getTaskIds(podMeta *statesinformer.PodMeta) {
+	pod := podMeta.Pod
+	containerMap := make(map[string]*corev1.Container, len(pod.Spec.Containers))
+	for i := range pod.Spec.Containers {
+		container := &pod.Spec.Containers[i]
+		containerMap[container.Name] = container
+	}
+
+	for _, containerStat := range pod.Status.ContainerStatuses {
+		container, exist := containerMap[containerStat.Name]
+		if !exist {
+			klog.Warningf("container %s/%s/%s lost during reconcile resctrl group", pod.Namespace,
+				pod.Name, containerStat.Name)
+			continue
+		}
+
+		containerDir, err := koordletutil.GetContainerCgroupParentDir(podMeta.CgroupDir, &containerStat)
+		if err != nil {
+			klog.V(4).Infof("failed to get pod container cgroup path for container %s/%s/%s, err: %s",
+				pod.Namespace, pod.Name, container.Name, err)
+			continue
+		}
+		ids, err := s.cgroupReader.ReadCPUTasks(containerDir)
+		if err != nil && resourceexecutor.IsCgroupDirErr(err) {
+			klog.V(5).Infof("failed to read container task ids whose cgroup path %s does not exists, err: %s",
+				containerDir, err)
+			continue
+		} else if err != nil {
+			klog.Warningf("failed to get pod container cgroup task ids for container %s/%s/%s, err: %s",
+				pod.Namespace, pod.Name, container.Name, err)
+			continue
+		}
+		if podMeta.ContainerTaskIds != nil {
```

Comment:
how about moving this check to the front

Reply:
Yes, let's do it.